clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -343,9 +350,30 @@ private KafkaAdminClient(AdminClientConfig config, String clientId, Time time, M
     }
 
     @Override
-    public void close() {
-        closed = true;
-        client.wakeup(); // Wake the thread, if it is blocked inside poll().
+    public void close(long waitTimeDuration, TimeUnit waitTimeUnit) {
+        long waitTimeMs = waitTimeUnit.toMillis(waitTimeDuration);
+        long now = time.milliseconds();
+        long newHardShutdownTimeMs = now + waitTimeMs;
```

Comment:
It may be worth handling overflow here as people could pass `Long.MAX_VALUE` for `waitTimeDuration`.